pkg/providers/cloudstack/cloudstack.go

```diff
@@ -530,13 +531,14 @@ func AnyImmutableFieldChanged(oldCsdc, newCsdc *v1alpha1.CloudStackDatacenterCon
 	if oldCsmc.Spec.ComputeOffering != newCsmc.Spec.ComputeOffering {
 		return true
 	}
-	if len(oldCsmc.Spec.UserCustomDetails) != len(newCsmc.Spec.UserCustomDetails) {
+	if !reflect.DeepEqual(oldCsmc.Spec.UserCustomDetails, newCsmc.Spec.UserCustomDetails) {
 		return true
 	}
-	for key, value := range oldCsmc.Spec.UserCustomDetails {
-		if value != newCsmc.Spec.UserCustomDetails[key] {
-			return true
-		}
+	if oldCsmc.Spec.Affinity != newCsmc.Spec.Affinity {
+		return true
+	}
+	if !reflect.DeepEqual(oldCsmc.Spec.AffinityGroupIds, newCsmc.Spec.AffinityGroupIds) {
+		return true
```

Comment:
can capc handle changing these attributes?

Reply:
No, capc doesn't allow them to be changed.